widgets/btc_linux.lua

```diff
@@ -0,0 +1,57 @@
+---------------------------------------------------
+-- Licensed under the GNU General Public License v2
+--  * (c) 2017, 0x5b <dragen15051@gmail.com>
+---------------------------------------------------
+
+-- {{{ Grab environment
+local setmetatable = setmetatable
+local pcall = pcall
+local helpers = require("vicious.helpers")
+local https = require("socket.http")
+local spawn = require("awful.spawn")
+local json = require("json")
+local string = {
+    sub = string.sub,
+    upper = string.upper,
+}
+-- }}}
+
+
+-- Btc: provides current bitcoin price
+-- vicious.widgets.btc
+local btc_linux = {}
+
+
+-- {{ Bitcoin widget type
+function btc_linux.async(format, warg, callback)
+    -- Default values
+    if not warg then warg = "usd" end
+
+    local btc = { ["{price}"] = "N/A" }
+    local currency_code = string.upper(warg)
+    local url = "https://api.coindesk.com/v1/bpi/currentprice/"..helpers.escape(currency_code)..".json"
+    local cmd = "curl --connect-timeout 1 "..url
+
+    -- {{ Checking response
+    local function parse(response)
+        local status, data = pcall(function() return json.decode(response) end)
+        if not status or not data then
+            return btc
+        end
+
+        btc["{price}"] = string.sub(data["bpi"][currency_code]["rate"], 0, -3)
+        return btc
+    end
+    -- }}
+
+    spawn.easy_async(cmd, function(stdout) callback(parse(stdout)) end)
+end
+-- }}}
+
+-- {{ Bitcoin widget type
+local function worker(format, warg)
```

Comment:
The synchronous variant would look a bit different, see `widgets/pkg_all.lua`. I think we do not need this function at all. `worker` and `setmetatable` can be removed here.

Just add `return worker` as the last statement.

Reply:
I dont understand it, nothing works without ```setmetatable```